tensorflow/python/framework/device_spec.py

```diff
@@ -356,6 +356,15 @@ def _components_to_string(job, replica, task, device_type, device_index):
     return output
 
   def __eq__(self, other):
+    """Checks if two known `DeviceSpec`s are same.
```

Comment:
I think to be more specific, since the DeviceSpec is kind of a composite class, the docstring probably should be:

"Return if the `other` DeviceSpec are same as the current instance, eg have same value for all the internal fields. "